intro-to-object-oriented-programming/defining-classes.md

```diff
@@ -0,0 +1,495 @@
+- **Must always take at least one parameter,** because it's special
+  - The first parameter **always** represents an instance itself. Details below.
+  - This parameter is typically named `self`
+- Does not need to return anything (and should not return anything)
+
+For example, consider this `Album` class, that will print a message whenever an `Album` is instantiated:
+
+```python
+class Album:
+
+    def __init__(self):
+        print("Hello! I print whenever a new Album is initialized")
+```
+
+<!-- Question 2 -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: Fshbd5
+* title: Constructor method
+
+##### !question
+
+Replace `pass` and add in the `__init__` method for this `Driver` class. In the `__init__` method, print out the string `"This init method runs every time a new Driver is instantiated!"`
+
+##### !end-question
+
+##### !placeholder
+
+```python
+class Driver:
+    pass
+```
+
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+from main import *
+
+class TestDriverClass(unittest.TestCase):
+  def test_driver_is_defined(self):
+    self.assertTrue(Driver())
+```
+
+##### !end-tests
+
+#### !explanation
+
+Actually, this test will pass as long as there's a `Driver` class; it's challenging to write a test that verifies that the `__init__` method exists.
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### !callout-info
+
+## "init" Is For Initialize
+
+We can describe the `__init__` method as a "class constructor," "constructor method," or "initializer." These names are used for any special method that runs once whenever an object is instantiated.
+
+### !end-callout
+
+## In Classes, We Can Access `self`
+
+In class definitions, in the `__init__` method and other instance methods, we will need to use a _special_ variable named `self`.
+
+Python wants to give instances the ability to reference itself with a variable. For example, as we write code to define the `Album` class, we may think about the "Dirty Computer" album. If we want to write code for "Dirty Computer" having its own state and behavior, we want a variable that represents "Dirty Computer" itself.
+
+This variable is almost always named `self`. Technically, this variable can have any name, but most Pythonists will use `self`.
+
+`self` must **always** be the first parameter to any instance method. `__init__` is an instance method. We'll see more examples of this when exploring instance methods.
+
+### !callout-info
+
+## An Analogy for `self`: The Pronoun "I"
+
+In the English language, there is the pronoun "I." Each person uses the pronoun "I" to describe themselves. However, "I" doesn't point to the same person in every context; "I" always means the person who is saying it!
+
+Exactly like that, the variable `self` will exist in every instance. However, the value of `self` will always mean the instance itself.
```

Comment:
```suggestion
## An Analogy for `self`: The Pronoun "I"

In the English language, there is the pronoun "I." Each person uses the pronoun "I" to describe themselves. However, "I" doesn't refer to the same person in every context. "I" always means the person who is saying it!

Exactly like that, the `self` parameter exists for every instance method. However, the value of `self` always means the current instance itself.
```

Fighting for the awkward "In the English language" bit -- I've hinted at this with other feedback, but when doing minor context switches (in this case, Python and OOP to the English language), I really enjoy using an awkward phrase that's very "IN YOUR FACE! SUBTLE CONTEXT SWITCH!" as a way to signal the reader to slow down and realize we're switching contexts.

Basically, "The English language has the pronount "I."" is very smooth, and I want it to be a little rough. I want the first thing the reader to think about is "The English language" and go "wait, uh, what do I think is the english language" and then see "there is the pronoun "I"".

"The English language has the pronoun "I."" comes across as so smooth, it comes across as a dense piece of information to take in, imo, and this isn't the place where we need dense information... we need the brain to be lifted into a kinda-weird space to make room for the analogy, not necessarily new technical information.

---

Minor and unrelated-- I was tempted to suggest

> However, "I" doesn't refer to the same person in every context- "I" always means the person who is saying it!

You had originally suggested a `.` instead of a `;`. Idk....... I'm still trying to learn how to write well, but I want some sort of connecting punctuation, not a full stop? This isn't very consequential so I won't fight for it :)

Reply:
We unofficially have spoken about ; before.  And I want to reiterate that every time I've seen you use it, it has been used correctly, and I love each and every one! My concern had been that a lot of people don't know how to use them, and can be confused at why there is a ; instead of a full stop.  Eventually over the course of this review, I acquiesced to your ; mastery and started letting them by. I think I eventually hit a sentence that was just too painful to change to a full stop. 😄 

And in retrospect, I do agree that the suggested opening does come off as very grammarphilic.  And I will do my best to bear in mind your desire to SHAKE THINGS UP at context switches. 😄